src/sync_session.cpp

```diff
@@ -230,7 +255,9 @@ struct sync_session_states::Dying : public SyncSession::State {
 
     void log_out(std::unique_lock<std::mutex>& lock, SyncSession& session) const override
     {
-        session.advance_state(lock, inactive);
+        if (session.m_pending_download_threads == 0 && session.m_pending_upload_threads == 0) {
+            session.advance_state(lock, inactive);
+        }
```

Comment:
earlier, we `advance_state` to `dying` if there are pending threads, why are we not doing that here?